src/main/java/com/ookawara/book/application/mapper/BookSqlProvider.java

```diff
@@ -40,12 +42,11 @@ public String updateBook(Book book) {
                 if (book.getIsPurchased() != null) {
                     SET("is_purchased = #{isPurchased}");
                 }
-                if (book.getCategoryId() != null) {
+                if (book.getCategoryId() >= 1) {
```

Comment:
不正な値なら IllegalArgumentException を throw するようにしてもいいかなと思います。
categoryID が 1 以上でないのは明らかに不正なリクエストなので。

Reply:
```suggestion
                if (book.getCategoryId() >= 1) {
                    SET("category_id = #{categoryId}");
                } else {
                    throw new IllegalArgumentException();
                }
```

このような形でIllegalArgumentException を throw するようにしてもよいということでしょうか？

Reply:
イメージあってます！